tests/mount-layer.bats

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bats
+
+load helpers
+
+@test "mount-layer" {
+	# Create a layer.
+	run storage --debug=false create-layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	layer="$output"
+
+	# Mount the layer.
+	run storage --debug=false mount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Unmount the layer.
+	run storage --debug=false unmount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" == "" ]
+
+	# Mount the layer twice.
+	run storage --debug=false mount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	run storage --debug=false mount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Unmount the third layer.
+	run storage --debug=false unmount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" == "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Unmount the third layer.
+	run storage --debug=false unmount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" == "" ]
+
+
+	# Mount the layer twice and force umount
+	run storage --debug=false mount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	run storage --debug=false mount $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Unmount the third layer.
+	run storage --debug=false unmount --force $layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	# Check it layer is mounted
+	run storage --debug=false mounted $layer
+	[ "$status" -eq 0 ]
+	[ "$output" == "" ]
+
+	# Try to delete the first layer, which should fail because it has children.
```

Comment:
This comment doesn't agree with the lines below it.

Reply:
Fixed